src/tools/doall/src/DOALL_chunking.cpp

```diff
@@ -118,6 +118,103 @@ void DOALL::rewireLoopToIterateChunks(LoopDependenceInfo *LDI) {
     this->IVValueJustBeforeEnteringBody[ivPHI] = chunkedIVValues;
   }
 
+  auto sccManager = LDI->getSCCManager();
+  auto sccdag = sccManager->getSCCDAG();
+
+  /*
+   * Generates code for periodic variable SCCs to match the DOALL chunking
+   * strategy.
+   */
+  for (auto scc : sccdag->getSCCs()) {
+    auto sccInfo = sccManager->getSCCAttrs(scc);
+    auto periodicVariableSCC = dyn_cast<PeriodicVariableSCC>(sccInfo);
+    if (periodicVariableSCC == nullptr)
+      continue;
+    // errs() << "DOALL: periodic variable with initial value " <<
+    // *periodicVariableSCC->getInitialValue() << "\n"; errs() << "       and
+    // period " << *periodicVariableSCC->getPeriod() << "\n"; errs() << " and
+    // step " << *periodicVariableSCC->getStepValue() << "\n";
+
+    /*
+     * Retrieve the relevant values of the periodic variable SCC.
+     */
+    auto initialValue = periodicVariableSCC->getInitialValue();
+    auto period = periodicVariableSCC->getPeriod();
+    auto step = periodicVariableSCC->getStepValue();
+    auto phi =
+        periodicVariableSCC->getPhiThatAccumulatesValuesBetweenLoopIterations();
+    auto taskPHI = cast<PHINode>(task->getCloneOfOriginalInstruction(phi));
+
+    auto entryBlock = (phi->getIncomingValue(0) == initialValue) ? 0 : 1;
+    auto loopBlock = 1 - entryBlock;
```

Comment:
It seems an overly complex logic. Can you simplify it perhaps using an if-then-else block?

Reply:
modified slightly in [557b357](https://github.com/arcana-lab/noelle/pull/74/commits/577b3570e8fec4c796a6d550d249cb6fcd689d34)